python/ray/experimental/serve/queues.py

```diff
@@ -220,3 +229,167 @@ def flush(self):
             self.self_handle._flush.remote()
         else:
             self._flush()
+
+
+class RandomPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for Random policy.
+    This backend selection policy is `Stateless` meaning
+    the current decisions of selecting backend are not
+    dependent on previous decisions.
+    """
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # while there are incoming requests and there are backends
+            while len(queue) and len(self.traffic[service]):
+                backend_names = list(self.traffic[service].keys())
+                backend_weights = list(self.traffic[service].values())
+                # randomly choose a backend for every query
+                chosen_backend = np.random.choice(
+                    backend_names, p=backend_weights).squeeze()
+
+                request = queue.popleft()
+                self.buffer_queues[chosen_backend].add(request)
+
+
+@ray.remote
+class RandomPolicyQueueActor(RandomPolicyQueue, CentralizedQueuesActor):
+    pass
+
+
+class RoundRobinPolicyQueue(CentralizedQueues):
+    """
+    A wrapper class for RoundRobin policy.
+    This backend selection policy is `Stateful` meaning
+    the current decisions of selecting backend are
+    dependent on previous decisions.
+    """
+
+    # Saves the information about last assigned
+    # backend for every service
+    round_robin_iter = {}
+
+    def set_traffic(self, service, traffic_dict):
+        logger.debug("Setting traffic for service %s to %s", service,
+                     traffic_dict)
+        self.traffic[service] = traffic_dict
+        backend_names = list(self.traffic[service].keys())
+        self.round_robin_iter[service] = itertools.cycle(backend_names)
+        self.flush()
+
+    def _flush_service_queue(self):
+        # perform traffic splitting for requests
+        for service, queue in self.queues.items():
+            # if there are incoming requests and there are backends
+            if len(queue) and len(self.traffic[service]):
+                if service not in self.round_robin_iter:
```

Comment:
Why would this be the case? Shouldn't set_traffic add them to the iterator already?

Reply:
This is the case if link method is used in queues.py

Reply:
I see. Can you change link to call `set_traffic`? so we can remove the redundancy.